source/library/Argo/Class/HasCodec.hs

```diff
@@ -132,11 +134,12 @@ instance HasCodec a => HasCodec (Object.Object a) where
                 $ Just (Nothing, either id Schema.Ref ref)
         }
 
+instance HasCodec a => HasCodec (Nullable.Nullable a) where
+    codec = nullable codec
+
 instance HasCodec a => HasCodec (Maybe a) where
     codec =
-        Codec.identified
-            $ Codec.mapMaybe (Just . Just) id codec
-            <|> Codec.map (const Nothing) (const $ Null.fromUnit ()) codec
+        Codec.identified $ Codec.map Nullable.toMaybe Nullable.fromMaybe codec
```

Comment:
This pattern comes up a lot for `newtype`s. Unfortunately the way it's set up you'll get separate definitions for both `Nullable a` and `Maybe a` if you use them both. Perhaps the `Maybe a` should just reference the other one?

``` js
{
  "definitions": {
    "Nullable Int": { /* ... */ },
    "Maybe Int": { "$ref": "#/definitions/Nullable Int" }
}
```